### PR TITLE
Fix invalid signers message broadcast on multikey

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- the consensus message with invalid signers was sent from self key instead of leader, which would not be compatible with multikey mode
  
## Proposed changes
- broadcast the consensus message from leader

## Testing procedure
- standard system test + multikey test

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by AdoAdoAdo — APPROVED** [review]

File: consensus/spos/bls/subroundEndRound.go [hunk 1]
```diff
@@ -589,20 +589,26 @@ func (sr *subroundEndRound) createAndBroadcastHeaderFinalInfo() {
 }
 
 func (sr *subroundEndRound) createAndBroadcastInvalidSigners(invalidSigners []byte) {
+	leader, errGetLeader := sr.GetLeader()
```
Comment (AdoAdoAdo) [inline consensus/spos/bls/subroundEndRound.go:592]: maybe also check if it should send the message in the context of a backup machine

Comment (sstanculeanu) [reply consensus/spos/bls/subroundEndRound.go:597]: pushed

File: consensus/spos/bls/subroundEndRound_test.go [hunk 2]
```diff
@@ -1556,29 +1603,56 @@ func TestVerifyInvalidSigners(t *testing.T) {
 func TestSubroundEndRound_CreateAndBroadcastInvalidSigners(t *testing.T) {
 	t.Parallel()
 
-	wg := &sync.WaitGroup{}
-	wg.Add(1)
+	t.Run("redundancy node should not send", func(t *testing.T) {
```
Comment (iulianpascalau) [inline consensus/spos/bls/subroundEndRound_test.go:1606]: "redundancy node should not send while the main is active"

**Review by iulianpascalau — APPROVED** [review]

**Review by iulianpascalau — DISMISSED** [review]

File: consensus/spos/bls/subroundEndRound.go [hunk 3]
```diff
@@ -589,20 +589,31 @@ func (sr *subroundEndRound) createAndBroadcastHeaderFinalInfo() {
 }
 
 func (sr *subroundEndRound) createAndBroadcastInvalidSigners(invalidSigners []byte) {
+	isSelfLeader := sr.IsSelfLeaderInCurrentRound() && sr.ShouldConsiderSelfKeyInConsensus()
+	if !isSelfLeader && !sr.IsMultiKeyLeaderInCurrentRound() {
```
Comment (AdoAdoAdo) [inline consensus/spos/bls/subroundEndRound.go:593]: can we change the condition to
```go
if !(isSelfLeader || sr.IsMultiKeyLeaderInCurrentRound()) 
```
would be easier to read

File: consensus/spos/bls/subroundEndRound_test.go [hunk 4]
```diff
@@ -1556,29 +1603,60 @@ func TestVerifyInvalidSigners(t *testing.T) {
...
+		wg.Add(1)
 
-	require.True(t, wasCalled)
+		expectedInvalidSigners := []byte("invalid signers")
+
+		wasCalled := false
+		container := mock.InitConsensusCore()
+		messenger := &mock.BroadcastMessengerMock{
+			BroadcastConsensusMessageCalled: func(message *consensus.Message) error {
+				wg.Done()
```
Comment (AdoAdoAdo) [inline consensus/spos/bls/subroundEndRound_test.go:1644]: can you do the wg.Done after the wasCalled = true?

**Review by iulianpascalau — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
Normal allin test: master-ace90ef010 -> fix_multikey_invalid_signe-624cbd54fe

--- Specific errors ---

block hash does not match 9031
wrong nonce in block 3592
miniblocks does not match 0
num miniblocks does not match 0
miniblock hash does not match 0
block bodies does not match 0
receipts hash missmatch 0

/------/


--- Statistics ---

Nr. of all ERRORS:  3
Nr. of all WARNS:  3319
Nr. of new ERRORS:  3
Nr. of new WARNS:  18
Nr. of PANICS:  0

/------/

--- ERRORS ---

/------/

**Pull request merged**